b2sdk/transferer/simple.py

```diff
@@ -22,10 +25,38 @@ def is_suitable(self, metadata, progress_listener):
 
     def download(self, file, response, metadata, session):
         actual_size = self._get_remote_range(response, metadata).size()
+        chunk_size = self._get_chunk_size(actual_size)
+
         digest = hashlib.sha1()
         bytes_read = 0
-        for data in response.iter_content(chunk_size=self._get_chunk_size(actual_size)):
+        for data in response.iter_content(chunk_size=chunk_size):
             file.write(data)
             digest.update(data)
             bytes_read += len(data)
+
+        # now normally bytes_read == metadata.content_length but sometimes there is a timeout
+        # or something and the server closes connection, while neither tcp or http have a problem
+        # with the truncated output, so we detect it here and try to continue
+
+        retries_left = self._num_tries - 1
+        while retries_left and bytes_read < metadata.content_length:
+            new_range = self._get_remote_range(response,
+                                               metadata).subrange(bytes_read, actual_size - 1)
```

Comment:
`actual_size - 1` would be wrong if `actual_size` were 0, but I think that can't happen because the body of this loop should never run for empty files.

Reply:
indeed. It's worth documenting in an `assert` though!